### PR TITLE
rustrs: Add rust driver crate

### PR DESCRIPTION
**Pull request by LorenzSchueler**
Add `driver` crate. This crate calls `evmrs` via the C FFI interface. It will be used for integration testing and for benchmarks.

### CODE REVIEW

**Comment by LorenzSchueler** [comment]
Note: The setup where the dev-dependencies of `evmrs` require `evmrs` is somewhat ugly but it is the only way I could find to enable additional features of the lib for integration tests because for integration tests the lib crate is compiled without the test feature.

File: rust/Cargo.toml [hunk 1]
```diff
@@ -1,15 +1,24 @@
+[workspace]
+members = ["driver"]
+
 [package]
 name = "evmrs"
 version = "0.1.0"
 edition = "2021"
 
+[features]
+# workaround for enabling mock feature also in integration tests
```
Comment (LuisPH3) [inline rust/Cargo.toml:10]: This comment could be formulated in a more descriptive way, for non-versed people to understand better
```suggestion
# Flag mock enables code generation for test mocking. This allows generating mocks for other build configuration but "test". 
```

File: rust/Cargo.toml [hunk 2]
```diff
@@ -1,15 +1,24 @@
...
+mock = ["dep:mockall"]
+
 [dependencies]
 bnum = "0.11.0"
 evmc-vm = { git = "https://github.com/Fantom-foundation/evmc", branch = "tosca-extensions" }
+mockall = { version = "0.13.0", optional = true }
 sha3 = "0.10.8"
 
 [dev-dependencies]
```
Comment (LuisPH3) [inline rust/Cargo.toml:19]: integration tests use dev-dependecies? how does this work?

File: rust/src/types/mock_execution_message.rs [hunk 3]
```diff
@@ -20,13 +18,17 @@ pub struct MockExecutionMessage<'a> {
     pub code: Option<&'a [u8]>,
 }
 
+impl<'a> MockExecutionMessage<'a> {
+    pub const DEFAULT_INIT_GAS: u64 = 1_000_000;
```
Comment (LuisPH3) [inline rust/src/types/mock_execution_message.rs:22]: :+1:  
better scoped, yes

**Review by LuisPH3 — APPROVED** [review]

File: rust/Cargo.toml (hunk 2 above)
Comment (LorenzSchueler) [reply rust/Cargo.toml:19]: yes. tests and benchmarks use additionally to the normal dependencies also dev-dependecies.

**Review by LuisPH3 — APPROVED** [review]

**Pull request merged**